mailmerge/template_message.py

```diff
@@ -102,6 +102,12 @@ def _make_message_multipart(self):
         # Create empty multipart message
         multipart_message = email.mime.multipart.MIMEMultipart('alternative')
 
+        original_encoding = str(self._message.get_charset())
+        # To prevent duplicate keys from being added to the message, remove
+        # these headers from the original message.
+        del self._message['Content-Type']
```

Comment:
Is L84 `_transform_encoding() ...` related to this problem?

Is the root cause of the problem that the outer message part has an encoding *and* the inner parts do as well?

Reply:
Yes, specifically because when we make a multipart message, we _copy over all headers_.

When we `_make_message_multipart`, we create a "multipart message", meaning that the content type of the outer message is set to `multipart/alternative`.

https://github.com/awdeorio/mailmerge/blob/a95c4951e91edac5c7bfa5e19f2389c7e4657d47/mailmerge/template_message.py#L102-L103

However, we also copy and preserve original headers. Since we always assign a content-type to non-multipart messages using `_transform_encoding()`, the content-type of the non-multipart message is also copied.

https://github.com/awdeorio/mailmerge/blob/a95c4951e91edac5c7bfa5e19f2389c7e4657d47/mailmerge/template_message.py#L105-L109

Reply:
Thanks for the clarification!